### DISCUSSION TITLE
Unable to conditionally set value.

### DISCUSSION THREAD

**Discussion by thecodejack** [data-driven-forms/react-forms#1001] [Need help] [answered]
Here is an example I am working on in codesandbox

https://codesandbox.io/s/poc-09f4i?file=/src/App.js

I am try to set `Gender` when `prefix` is updated. Somehow this not working. Not sure if I did something wrong or problem with Material components.

**Comment by rvsia** [comment, marked as answer]
Hello, `value` on the select is wrong. Use [initialValue](https://data-driven-forms.org/schema/introduction#initialvalue). Then it will work correctly.

Value in [FinalForm](https://final-form.org/docs/react-final-form/types/FieldProps#value) is used only with checkboxes/radios.

**Reply by thecodejack** [reply]
This worked. Thanks.
